manifests/profile/yarn.pp

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2020 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+#
+# nebula::profile::yarn
+#
+# Installs Yarn from Official repositories.
+class nebula::profile::yarn (
+) {
+
+  apt::source { 'yarn':
+    location     => 'https://dl.yarnpkg.com/debian/',
+    key          =>  {
+      id     => 'B90F6449FEBC20F00DB13ED8212659B22565CA86',
+      source => 'https://dl.yarnpkg.com/debian/pubkey.gpg'
+    },
+    release      => 'stable',
+    repos        => 'main',
+    architecture => $::os['architecture'],
+  }
+
+  package { ['yarn']: }
```

Comment:
Might be worth adding a requirement here, since the package requires the apt repo, like this

```puppet
package { 'yarn':
  require => Apt::Source['yarn'],
}
```

This will ensure that the puppet agent doesn't try to apt-get install yarn until after it's added the new repo.

Reply:
Ah yes, we did talk about this.  Thanks for the example.  I agree, I like this approach.  I will change it.